1/backend/src/resolvers.js

```diff
@@ -0,0 +1,86 @@
+const jwt = require('jsonwebtoken');
+const fs = require('fs');
+
+data = require('./data');
+
+let SECRET_KEY = fs.readFileSync('./src/key/secret.key', 'utf8');
+
+const resolvers = {
+    Query: {
+        todos: () => JSON.parse(JSON.stringify(data.todos))
+    },
+    Mutation: {
+        addTodo: (parent, args, context) => {
+            if(args.message != "" && args.message != null && context.token) {
+                let newTodo = {
+                    id: findNextId(),
+                    message: args.message,
+                    completed: false,
+                };
+                data.todos.push(newTodo);
+                return newTodo;
+            }
+            return;
+        },
+        deleteTodo: (parent, args, context) => {
+            if (args.id != null && context.token){
+                let index = data.todos.findIndex(td => td.id === parseInt(args.id));
+                if (index > 0){
+                    data.todos.splice( index, 1)
+                    return true
+                }
+                return false
+            }
+        },
+        editTodo: (parent, args, context) => {
+            if (args.id != null && args.message != null && context.token){
+                tod = data.todos.find(td => td.id === parseInt(args.id))
+                if (tod){
+                    tod.message = args.message;
+                    return tod;
+                }
+                return
+            }
+
+        },
+        finishTodo: (parent, args, context) => {
+            if (args.id != null && context.token){
+                let tod = data.todos.find(td => td.id === parseInt(args.id));
+                if (tod){
+                    //Reverse status of item
+                    tod.completed = !tod.completed;
```

Comment:
Then I wouldn't call it `finishTodo`. That would imply you can only set it from `pending` to `completed`, not vice versa. How about `toggleCompleted`?